### PR TITLE
80367 add prefill 10 7959f 1

### PR DESCRIPTION
**Pull request by balexandr**
Add back-end logic to enable prefill for vets-website. This is an update to the original PR that could be reverted in https://github.com/department-of-veterans-affairs/vets-api/pull/16481

## Summary

- Added 10-7959F-1 for FormProfile

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/80367

## Testing done

- Can't add normal tests as we are moving from vets-json-schema.
https://github.com/department-of-veterans-affairs/VA.gov-team-forms/blob/main/Engineering/Form%20Digitization%20development%20guide.md#step-7-collaborative-add-authenticated-experience-support-to-the-form
Screenshot 2024-04-25 at 1 17 48 PM
- Added a custom prefill test

## What areas of the site does it impact?
- Nothing live

### CODE REVIEW

**Comment by balexandr** [comment]
@RachalCassity The other ticket hasn't been reverted yet so this only has the changes to the mapping. I added a comment in the desc as to why there are no tests relating to this.

**Comment by balexandr** [comment]
> Please provide testing in [Form Profile Spec](https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/models/form_profile_spec.rb)

@RachalCassity See in the desc about adding test to form_profile_spec. It's advised to omit the test as we are moving away from vets-json-schema.

File: config/form_profile_mappings/10-7959F-1.yml [hunk 1]
```diff
@@ -1,22 +1,17 @@
 
 veteran:
-  date_of_birth: [identity_information, date_of_birth]
-  full_name: [identity_information, full_name]
+  dateOfBirth: [identity_information, date_of_birth]
+  fullName: [identity_information, full_name]
   first: [identity_information, first]
```
[the remark this answers is not in the document]
Comment (balexandr) [reply config/form_profile_mappings/10-7959F-1.yml]: Removed first, middle and last.

**Review by michaelclement — COMMENTED** [review]
lgtm

**Review by reiting — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**